### PR TITLE
feat(connector): [Authorizedotnet] Use connector_response_reference_id as reference to merchant

### PR DESCRIPTION
**Pull request by amitroy-thedev**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by srujanchikke** [comment]
Hey @amitroy-thedev ,

Please use the following cargo commands -

`cargo check --all-features`                                                                 
`cargo clippy --all-features` 

ref -[ Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands), and fix the errors thrown by clippy for the checks to pass .

Please run `cargo +nightly fmt`  to fix failing formatting check

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 1]
```diff
@@ -604,7 +604,7 @@ impl<F, T>
                             mandate_reference: None,
                             connector_metadata: metadata,
                             network_txn_id: transaction_response.network_trans_id.clone(),
-                            connector_response_reference_id: None,
+                            connector_response_reference_id: Some(ref_trans_id.to_string()),
```
Comment (srujanchikke) [inline crates/router/src/connector/authorizedotnet/transformers.rs:607]: make the same changes here

**Review by srujanchikke — APPROVED** [review]

**Comment by srujanchikke** [comment]
Please resolve the conflicts @amitroy-thedev

**Comment by srujanchikke** [comment]
Hey @amitroy-thedev ,

#2292 is regarding `connector_request_reference_id` not `connector_response_reference_id` . Your changes have been already done in #2497 . Could make changes in payment request . Please reach out to us if you have any other question understanding the problem statement .

**Comment by srujanchikke** [comment]
`refId` would be the merchant reference id according to api docs of [authroizedotnet](https://developer.authorize.net/api/reference/index.html#payment-transactions-authorize-a-credit-card) . You need to declare `refId` in `AuthorizedotnetPaymentsRequest` struct and map to `connector_request_reference_id` .

**Comment by amitroy-thedev** [comment]
@srujanchikke
Sorry for the mistake let me try once again by make a new repo again

**Pull request closed without merging**